fix: Adapt packaging to the new sha addressing

craciunoiuc opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [ ] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [ ] Tested your changes against relevant architectures and platforms;
  - [ ] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

craciunoiuc committed:
fix(internal): Check package version also

To check if package is 100% missing a version

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

nderjung reviewed (commented):
Shouldn't this be `@sha256:` ?

This might nor guarantee the right version anymore being deployed.

craciunoiuc committed:
fix(oci): Use index when generating package ID for now

Only the index workflow is supported right now. When we will handle
manifest-only workflows, this will need updating.

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

nderjung reviewed (approved):
Thanks!

Reviewed-by: Alexander Jung <alex@unikraft.com>
Approved-by: Alexander Jung <alex@unikraft.com>
